isis/src/mro/apps/marci2isis/main.cpp

```diff
@@ -153,9 +154,83 @@ void IsisMain() {
     isisCube.findGroup("Instrument").addKeyword(PvlKeyword("Framelets", framelets[i]));
 
     outputCubes[i]->write(origLabel);
-    delete outputCubes[i];
   }
 
+  QString prodId = outputCubes[0]->label()->findGroup("Archive", Pvl::Traverse)["ProductId"][0]; 
+  prodId = prodId.toUpper();
+  vector<int> frameseq;
+  vector<double> exptime;
+  // Populate with first values
+  Pvl *isisLabelInitial = outputCubes[0]->label();
+  PvlGroup &instInitial = isisLabelInitial->findGroup("Instrument", Pvl::Traverse);
+  double exposure = instInitial["ExposureDuration"][0].toDouble() * 1000.0;
+  frameseq.push_back(0);
+  exptime.push_back(exposure); 
+
+  QString varExpFile = "$mro/calibration/marci/varexp.tab";
+  
+  // Load the MARCI exposure duration calibration tables.
+  bool header=false;
+  int skip=0;
+  FileName csvfile(varExpFile);
+    
+  CSVReader csv(csvfile.expanded(), header, skip);
+    // There may be multiple entries in the file for this productID,
+    // so we *must* loop through the entire file.
+  for(int i = 0 ; i < csv.rows() ; i++) {
+      CSVReader::CSVAxis row = csv.getRow(i);
+      // The productId in the file is encapsulated by double quotes.
+      QString fileProdId = row[0];
+      // This is garbage code, but my compiler isn't allowing me to escape the double quotes
+      int prodIdLastIndex = fileProdId.size() - 1 ;
+      fileProdId.remove(prodIdLastIndex,1);
+      fileProdId.remove(0,1);
+      // Now, compare product ids from the input image and from the calibration table.
+      if(fileProdId == prodId ) {
+        if((row.dim1() - 1) != 2) {
+          QString msg = "This appears to be a malformed calibration file."; 
+                  msg += " There are not enough columns in the CSV";
+                  msg += " file to perform the exposure time correction.";
+          throw IException(IException::User, msg, _FILEINFO_);
+        }
+        // Build the two vectors, exptime and frame. We'll relate those to each other
+        // back in main(). Remember that a productID may have multiple entries in the table.
+        frameseq.push_back(toInt(row[1]));
+        exptime.push_back(toDouble(row[2]));
+      }
+  }
+
+  if (flip && exptime.size() > 0) {
+    reverse(frameseq.begin(),frameseq.end());
+    reverse(exptime.begin(),exptime.end());
```

Comment:
I flipped these to be consistent with `marcical`, but maybe it's not actually necessary @scsides, thoughts?

Reply:
If the marcical update was written expecting this to be flipped, then I'm okay with flipping it here. It can be addressed again as part of a broader look at the flipping.

Reply:
I can go either way on this one, but marcical needs to know which one it will be.

Reply:
again keeping current to make marcical work simpler